src/Symfony/Component/ErrorHandler/Resources/stubs/Debug.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Debug;
+
+@trigger_error(sprintf('The "%s" class is deprecated since Symfony 4.4, use "%s" instead.', Debug::class, \Symfony\Component\ErrorHandler\Debug::class), E_USER_DEPRECATED);
```

Comment:
when is this caught and reported to the user?

Reply:
only when `Symfon\Component\Debug\Debug` class is used

Reply:
I mean: when is it displayed on a screen or logged to a file? Can you do it?

Reply:
people can get rid of it by changing to `Symfony\Component\ErrorHandler\Debug` in their front

Reply:
Sorry, I really mean: when is the deprecation presented to the user so that they can read about it? Isn't this totally silent at the moment? There is no error handler registered yet when this is called, so that this goes to /dev/null. I confirmed, then this fails to be useful and we go back to start. Right?

Reply:
That's another matter to solve I think, because that'll happen even without this patch having only the legacy Debug component installed. Deprecation warnings on legacy Debug, ErrorHandler, etc classes are totally silent at the moment, yes, there is no error handler registered yet when they're called.

Reply:
It's a chicken and egg situation. We cannot fix this, so that's not another matter to me: if this cannot be caught, it should be removed and we must decide on an alternate strategy.

Maybe we can agree that the annotation on the stub class is enough for IDEs/static analyzers to work, and we shouldn't try harder, because we simply can't.

I'd be OK with that personally. But let's not hide from the truth :)

Reply:
I agree we don't need a real deprecation warning for this. But IMO we can just remove the alias in 5.0. There is no point in delaying the cut in 6.0 when we can't trigger a real deprecation anyway. So to me the alias should only be present in 4.4.

Reply:
@Tobion works for me.

Reply:
So we all good on this alternative?

Reply:
yes, once this line is removed :)

Reply:
Done.